src/gui/editors/PianoRoll.cpp

```diff
@@ -3261,8 +3287,22 @@ int PianoRoll::getKey(int y ) const
 	return key_num;
 }
 
+QList<int> PianoRoll::getAllOctavesForKey( int keyToMirror ) const
+{
+	QList<int> keys;
 
+	for (int i = 0; i < NumKeys; ++i)
+	{
+		if ((i < keyToMirror && (keyToMirror - i) % KeysPerOctave == 0) || (i > keyToMirror && (i - keyToMirror) % KeysPerOctave == 0))
```

Comment:
This conditional can be reduced to `if ((i - keyToMirror) % KeysPerOctave == 0 && i != keyToMirror)`.

It seems to me that the whole function could even be simplified down to:

```c++
QList<int> keys;
for (int i=keyToMirror % KeysPerOctave; i < NumKeys; i += KeysPerOctave)
{
    keys.append(i);
}
return keys;
```

It's *almost* identical, except that `keyToMirror` appears in the middle of the list instead of at the end. It doesn't seem like that would matter, but maybe it does? If it does matter, then I would propose this version:

```c++
QList<int> keys;
for (int i=keyToMirror % KeysPerOctave; i < NumKeys; i += KeysPerOctave)
{
    if (i != keyToMirror)
    {
        keys.append(i);
    }
}
keys.append(keyToMirror);
return keys;
```

This behaves identically to your current version, but is more efficient and, in my opinion, easier to understand.